pkg/strategy/grid2/active_order_recover.go

```diff
@@ -0,0 +1,151 @@
+package grid2
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/exchange/retry"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/c9s/bbgo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"go.uber.org/multierr"
+)
+
+type SyncActiveOrdersOpts struct {
+	logger            *logrus.Entry
+	metricsLabels     prometheus.Labels
+	activeOrderBook   *bbgo.ActiveOrderBook
+	orderQueryService types.ExchangeOrderQueryService
+	exchange          types.Exchange
+}
+
+func (s *Strategy) initializeRecoverCh() bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	alreadyInitialize := false
+
+	if s.activeOrdersRecoverCh == nil {
+		s.logger.Info("initialize recover channel")
+		s.activeOrdersRecoverCh = make(chan struct{}, 1)
+	} else {
+		s.logger.Info("already initialize recover channel, trigger active orders recover")
+		alreadyInitialize = true
+		s.activeOrdersRecoverCh <- struct{}{}
```

Comment:
use select + default case to skip it if the channel is full?

Reply:
OK!